src/service.rs

```diff
@@ -347,297 +338,34 @@ pub fn new_light(config: NodeConfiguration)
 				Some(Box::new(finality_proof_import)),
 				client.clone(),
 				inherent_data_providers.clone(),
+				spawn_task_handle,
+				registry,
 			)?;
 
 			Ok((import_queue, finality_proof_request_builder))
 		})?
-		.with_finality_proof_provider(|client, backend|
-			Ok(Arc::new(GrandpaFinalityProofProvider::new(backend, client)) as _)
-		)?
+		.with_finality_proof_provider(|client, backend| {
+			// GenesisAuthoritySetProvider is implemented for StorageAndProofProvider
+			let provider = client as Arc<dyn StorageAndProofProvider<_, _>>;
+			Ok(Arc::new(GrandpaFinalityProofProvider::new(backend, provider)) as _)
+		})?
 		.with_rpc_extensions(|builder,| ->
-			Result<RpcExtension, _>
-		{
-            use substrate_frame_rpc_system::{LightSystem, SystemApi};
-
-            let client = builder.client().clone();
-            let pool = builder.pool();
-			let fetcher = builder.fetcher()
-				.ok_or_else(|| "Trying to start node RPC without active fetcher")?;
-			let remote_blockchain = builder.remote_backend()
-				.ok_or_else(|| "Trying to start node RPC without active remote blockchain")?;
-
-            let mut io = jsonrpc_core::IoHandler::default();
-            io.extend_with(
-                SystemApi::<AccountId, Index>::to_delegate(LightSystem::new(client, remote_blockchain, fetcher, pool))
-            );
-            Ok(io)
-        })?
-		.build()?;
-
-	Ok(service)
-}
-
-#[cfg(test)]
```

Comment:
it was removed from the source

Reply:
I still see them in `bin/node/cli/src/service.rs`

Reply:
Could you double-check?

Reply:
those two tests are not added since they are ignored and previous tests that were not ignored have been removed altogether. I would rather not include them only to be ignored by the test runner

Reply:
Okay, sounds good!